mlflow/onnx.py

```diff
@@ -182,10 +182,27 @@ def _load_model(model_file):
 
 
 class _OnnxModelWrapper:
-    def __init__(self, path):
+    def __init__(self, path, providers=None):
         import onnxruntime
 
-        self.rt = onnxruntime.InferenceSession(path)
+        if providers is None:
+            providers = ['CUDAExecutionProvider', 'CPUExecutionProvider']
+
+        # NOTE: Some distributions of onnxruntime require the specification of the providers argument on calling. E.g. onnxruntime-gpu.
+        #       The package import call does not differnetiate which architecture specific version has been installed, as all are imported with onnxruntime.
+        #       onnxruntime documentation says that from v1.9.0 some distributions require the providers list to be provided on calling an InferenceSession.
+        #       Therefore the try catch nested structure below attempts to create an inference session with just the model path as pre v1.9.0.
+        #       If that fails, it will use the providers list call. 
```

Comment:
@ecm200 Would it be possible to always instantiate the InferenceSession with `providers`, or should `providers` only be used as a fallback?

Reply:
@dbczumar That’s a very good point and only occurred to me last night.

Certainly v1.8.0 does have the variable but it is not required. I am not sure how far back the variable is present in the `onnxruntime` releases before that.

Potentially I think it’s easier to define it this way, as it’s a clearly defined breaking change in the `onnxruntime` documentation.

I suppose we could just pin to a minimum version of `onnxruntime` to `mlflow`?

So I’m erring to the current implementation but happy to do what you admins think is more appropriate or online with `mlflow` style and policy.

Reply:
@ecm200 Great point! In the interest of supporting a broader set of onnxruntime versions, I'm also in favor of the current implementation. Thank you for the detailed explanation.

Reply:
@dbczumar  New commit implements refactored approach as discussed.